### DISCUSSION TITLE
recommended settings Zettelkasten

### DISCUSSION THREAD

**Discussion by jeom123** [brianpetro/obsidian-smart-connections#458] [General]
Which embeddings model and settings would you recommend for someone that want to use the plugin on a Zettelkasten note folder? I want accuracy, ready to pay, have a GPT API, etc etc.

Notes embedding model?
Blocks embedding model?
Smart chat model?

I have tried putting everything at the "highest" (i.e. bottom) setting, but then the notes does not seem to index and I get "API error" when I try smart chat. The default setting worked, but would I get better accuracy with another model?

Thanks!

**Comment by brianpetro** [comment]
Hi @jeom123 

If you only have short notes, with few headings per note, then note-level embeddings may be sufficient (set the the blocks to none).

If you're new to the plugin, which embedding model you use won't make a big difference.

Not sure what you mean by highest. Can you take screenshots?

Also, if you can open the developer console (View > Toggle Developer Tools) to screenshot any errors that occur when you get the API error, that would be helpful.

Thanks for trying out Smart Connections 😊

🌴

**Reply by jeom123** [reply]
I mean that I have simply chosen the last/bottom option in the dropdown lists (Ada), sensing that this would be the most powerful one :)
In any case, my Zettelkasten notes consist of a few headings, some text, but probably more than most that use this system (I copy-paste some supporting stuff from reading notes). So maybe 300-800 words per note. Which settings do you recommend for note- and block-embeddings in this case?

**Reply by brianpetro** [reply]
@jeom123 this is my setup 
![Screenshot_20240214-084018](https://github.com/brianpetro/obsidian-smart-connections/assets/1886014/f57f3618-12a1-4266-ac66-82427329b1a8)

Ada is older than the OpenAI small/large models, so you might want to try either of them. And the small/large models with few dimensions have been shown to work as well as Ada, with few resources. The higher dimension ones should be more powerful, but whether that will have a significant impact on Smart Connections is yet TBD. 

🌴

**Reply by jeom123** [reply]
Thanks! I will try this out.

**Comment by mccleanla** [comment]
hi there. struggling the exact same way wiht trying to follow the instructions and it not working.   I have a paid openAI account, but getting the api key is invalid.  no idea what to choose in the various boxes in obsidian, but various combos tried and nothing works... :-/

**Reply by brianpetro** [reply]
When you add the OpenAI API key and hit "save" , a notification should pop-up that tells you whether the API key is valid or not.

**Reply by Richie-Peak** [reply]
mccleanla not to be patronising, but have you tried deleting your api key and making a new one?

**Comment by mccleanla** [comment]
It says not valid. But I literally copied/pasted from one window to another. It is valid…On 13 Feb 2024, at 16:40, WFH Brian ***@***.***> wrote:﻿
When you add the OpenAI API key and hit "save" , a notification should pop-up that tells you whether the API key is valid or not.

**Reply by brianpetro** [reply]
@mccleanla 
There are currently some API outages at OpenAI, so maybe that's the issue https://status.openai.com/

🌴

**Reply by brianpetro** [reply]
@mccleanla I'm pretty sure it's related to the current outages https://status.openai.com/

I'm getting the same error now, too.

**Reply by jeom123** [reply]
I also got this error - until I topped up my OpenAI account ;)

**Comment by mccleanla** [comment]
Awesome… ill try again when its sortedOn 13 Feb 2024, at 20:33, WFH Brian ***@***.***> wrote:﻿
@mccleanla I'm pretty sure it's related to the current outages https://status.openai.com/
I'm getting the same error now, too.

**Comment by mccleanla** [comment]
Only a little patronising ;)Yes. On 29 Feb 2024, at 12:25, Thomas Richardson ***@***.***> wrote:﻿
mccleanla not to be patronising, but have you tried deleting your api key and making a new one?
